cinema/serializers.py

```diff
@@ -68,13 +80,91 @@ class Meta:
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available",
         )
 
+    def get_tickets_available(self, obj):
+        taken_places_count = Ticket.objects.filter(movie_session=obj).count()
+        return obj.cinema_hall.capacity - taken_places_count
+
+
+class TicketSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ticket
+        fields = ["id", "row", "seat", "movie_session"]
+
+
+class TicketListSerializer(TicketSerializer):
+    class Meta:
+        model = Ticket
+        fields = ("row", "seat")
+
 
 class MovieSessionDetailSerializer(MovieSessionSerializer):
     movie = MovieListSerializer(many=False, read_only=True)
     cinema_hall = CinemaHallSerializer(many=False, read_only=True)
+    taken_places = TicketListSerializer(
+        many=True,
+        read_only=True,
+        source="tickets"
+    )
 
     class Meta:
         model = MovieSession
-        fields = ("id", "show_time", "movie", "cinema_hall")
+        fields = ("id", "show_time", "movie", "cinema_hall", "taken_places")
+
+
+class TicketDetailSerializer(TicketSerializer):
+    movie_session = MovieSessionListSerializer(many=False, read_only=True)
+
+    class Meta:
+        model = Ticket
+        fields = ("id", "row", "seat", "movie_session")
+
+
+class OrderSerializer(serializers.ModelSerializer):
```

Comment:
it seems to me, u could use orderlistserializer for inheriting here

Reply:
I fixed the two previous problems. I don't understand here. you mean like this:
 **class OrderSerializer(OrderListSerializer):**

Reply:
yeah, orderlist and orderserializer has the same meta, the only thing u need to change is create method and tickets attr, but it is not crucial